operator/src/test/java/org/keycloak/operator/RealmImportE2EIT.java

```diff
@@ -0,0 +1,153 @@
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.Test;
+import org.keycloak.operator.v2alpha1.crds.KeycloakRealmImport;
+import org.keycloak.operator.v2alpha1.crds.KeycloakRealmStatusCondition;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.keycloak.operator.v2alpha1.crds.KeycloakRealmStatusCondition.*;
+
+@QuarkusTest
+public class RealmImportE2EIT extends ClusterOperatorTest {
+
+    final static String KEYCLOAK_SERVICE_NAME = "example-keycloak";
+    final static int KEYCLOAK_PORT = 8080;
+
+    private KeycloakRealmStatusCondition getCondition(List<KeycloakRealmStatusCondition> conditions, String type) {
+        return conditions
+                .stream()
+                .filter(c -> c.getType().equals(type))
+                .findFirst()
+                .get();
+    }
+
+    @Test
+    public void testWorkingRealmImport() {
+        Log.info(((operatorDeployment == OperatorDeployment.remote) ? "Remote " : "Local ") + "Run Test :" + namespace);
+        // Arrange
+        k8sclient.load(getClass().getResourceAsStream("/example-postgres.yaml")).inNamespace(namespace).createOrReplace();
+        k8sclient.load(getClass().getResourceAsStream("/example-keycloak.yml")).inNamespace(namespace).createOrReplace();
+
+        k8sclient.services().inNamespace(namespace).create(
+                new ServiceBuilder()
+                        .withNewMetadata()
+                        .withName(KEYCLOAK_SERVICE_NAME)
+                        .withNamespace(namespace)
+                        .endMetadata()
+                        .withNewSpec()
+                        .withSelector(Map.of("app", "keycloak"))
+                        .addNewPort()
+                        .withPort(KEYCLOAK_PORT)
+                        .endPort()
+                        .endSpec()
+                        .build()
+        );
+
+        // Act
+        k8sclient.load(getClass().getResourceAsStream("/example-realm.yaml")).inNamespace(namespace).createOrReplace();
+
+        // Assert
+        Awaitility.await()
+                .atMost(3, MINUTES)
+                .pollDelay(5, SECONDS)
+                .ignoreExceptions()
+                .untilAsserted(() -> {
+                    var conditions = k8sclient
+                            .resources(KeycloakRealmImport.class)
+                            .inNamespace(namespace)
+                            .withName("example-count0-kc")
+                            .get()
+                            .getStatus()
+                            .getConditions();
+
+                    assertThat(getCondition(conditions, DONE).getStatus()).isTrue();
+                    assertThat(getCondition(conditions, STARTED).getStatus()).isFalse();
```

Comment:
I'd like this test to be extended a little bit. We should test that the CR goes through `started` as well, not just directly wait for `done`.

Reply:
ok 👍